File: src/app/core/interfaces/usecases/imotorista-usecase.ts
import { Observable } from 'rxjs';
import { MotoristaModel } from '../../domain/entity/motorista-model';

export abstract class IMotoristaUsecase {
  abstract get(id?: number): Observable<MotoristaModel>;
  abstract insert(param: MotoristaModel): Observable<MotoristaModel>;
  abstract update(param: MotoristaModel): Observable<MotoristaModel>;
  abstract delete(id: number): Observable<boolean>;
}

File: mobile/lib/src/app/interfaces/usecases/imotorista_usecase.dart
import 'package:truckpad/src/app/domain/entity/motorista_entity.dart';

abstract class IMotoristaUsecase {
  get(int id);
  insert(MotoristaEntity param);
  update(MotoristaEntity param);
  disableEnable(int id, bool status);
}